tests/unit/parallel/algorithms/remove_copy_if.cpp

```diff
@@ -23,8 +23,8 @@ void test_remove_copy_if(ExPolicy const& policy, IteratorTag)
 
     std::vector<int> c(10007);
     std::vector<int> d(c.size());
-    auto middle = boost::begin(c) + c.size()/2;
-    std::iota(boost::begin(c), middle, std::rand());
+    auto middle = boost::begin(c) + std::rand() % (c.size()/2);
+    std::iota(boost::begin(c), middle, static_cast<int>(std::rand() % c.size()));
```

Comment:
The first change (line 26) does not really fix anything, but rather increases the randomness of the tests by modifying the point from which on the input data is reset.

The second change (line 27) avoids that iota generates negative values (caused by overflowing before performing 5003 increments). This was causing the test to occasionally fail.